crates/red_knot_python_semantic/src/types/subclass_of.rs

```diff
@@ -96,8 +96,59 @@ impl<'db> SubclassOfType<'db> {
 
     pub(crate) fn to_instance(self) -> Type<'db> {
         match self.subclass_of {
-            ClassBase::Class(class) => Type::instance(class),
-            ClassBase::Dynamic(dynamic_type) => Type::Dynamic(dynamic_type),
+            SubclassOfInner::Class(class) => Type::instance(class),
+            SubclassOfInner::Dynamic(dynamic_type) => Type::Dynamic(dynamic_type),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, salsa::Update)]
+pub(crate) enum SubclassOfInner<'db> {
+    Class(ClassType<'db>),
+    Dynamic(DynamicType),
+}
```

Comment:
Add a doc comment here, explaining that it's similar to `ClassBase` but only includes the variants that are valid in a subclass-of type